### PR TITLE
Better handling http monitoring is disabled error

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?

Better handling `http monitoring is disabled` error

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/protocols/http/monitor.go [hunk 1]
```diff
@@ -27,7 +27,8 @@ import (
 )
 
 var (
-	startupError error
+	MonitorIsDisabledError = fmt.Errorf("http monitoring is disabled")
+	startupError           error
```
[the remark this answers is not in the document]
Comment (guyarb) [reply pkg/network/protocols/http/monitor.go]: great idea, added

File: pkg/network/protocols/http/monitor.go [hunk 2]
```diff
@@ -149,7 +162,12 @@ func (m *Monitor) Start() error {
 }
 
 func (m *Monitor) GetUSMStats() map[string]interface{} {
-	if m == nil {
+	if m == nil || state != Running {
```
Comment (hmahmood) [inline pkg/network/protocols/http/monitor.go:165]: You could just return it all: 
```go
map[string]interface{}{
  "last_check": m.telemtry.then,
  "state": state,
  "error": startupError,
}
```

Comment (guyarb) [reply pkg/network/protocols/http/monitor.go:165]: can't do that like that, modified the suggestion

**Review by hmahmood — APPROVED** [review]

**Comment by hmahmood** [comment]
Do you want to modify the state template as well?

**Comment by p-lambert** [comment]
I'm fine with merging this as it is, but another possible approach would be having a `http.Monitor` *interface* and return a no-op implementation (instead of relying on the `nil` value) which` would store the state/error in these special cases.
I'm my opinion the main benefit of this would be no longer relying on global variables (and not having to "capture" the error)

**Comment by guyarb** [comment]
> I'm fine with merging this as it is, but another possible approach would be having a `http.Monitor` _interface_ and return a no-op implementation (instead of relying on the `nil` value) which would just store the state/error in these special cases. I'm my opinion the main benefit of this would be no longer relying on global variables (and not having to "capture" the error)
> 
> edit: in fact we already have that interface for Windows for some reason, so we could probably just move that to a file used by all builds:
> 
> https://github.com/DataDog/datadog-agent/blob/26ac5ef74b13de260662c8782ac4ddeab0c33889/pkg/network/protocols/http/monitor_windows.go#L21-L26
> 
> having the no-op implementation would also make it easier for windows to handle initialization errors or the disabled case

What do you mean by no-op implementation?
do you mean that every method of httpMonitor will check if the state is `Running` before executing anything?

**Comment by p-lambert** [comment]
@guyarb I mean something like this

```go
var _ http.Monitor = &noopImpl{} //noopImpl implements http.Monitor

type noopImpl struct {
	err error
	state int
}

func (n *noopImpl) Start() {}
func (n *noopImpl) GetHTTPStats() map[Key]*RequestStats { return nil }
func (n *noopImpl) Stop() error { return nil }
func (n *noopImpl) GetUSMStats() (map[string]interface{}) {
	return map[string]int64{
		"state": n.state,
		"error": n.err,
	}
}
```

And then inside `NewHTTPMonitor`, you could write something like this:
```
if !c.EnableHTTPMonitoring {
	return &noopImpl{fmt.Errorf("http monitoring is disabled"), NotRunning}
}
```

**Review by sgnn7 — APPROVED** [review]
LGTM for ASC-owned `pkg/status/templates/systemprobe.tmpl`. Lots of failures here too so I'm guessing it needs a rebase.

**Comment by guyarb** [comment]
> @guyarb no, I mean something like this
> 
> ```go
> var _ http.Monitor = &nullMonitor{} //nullMonitor implements http.Monitor
> 
> type nullMonitor struct {
> 	err error
> 	state int
> }
> 
> func (n *nullMonitor) Start() {}
> func (n *nullMonitor) GetHTTPStats() map[Key]*RequestStats { return nil }
> func (n *nullMonitor) Stop() error { return nil }
> func (n *nullMonitor) GetUSMStats() (map[string]interface{}) {
> 	return map[string]int64{
> 		"state": n.state,
> 		"error": n.err,
> 	}
> }
> ```
> 
> And then inside `NewHTTPMonitor`, you could write something like this:
> 
> ```go
> if !c.EnableHTTPMonitoring {
> 	return &nullMonitor{fmt.Errorf("http monitoring is disabled"), Disabled}
> }
> ```

Not sure it is a good usage here. We are going to merge this week http2 and kafka decoding. Those PRs will definitely add more methods in the interface which are not supported by the windows agent. Thus, I have to have a different interface, just to implement the no-op instance, which seems to me a bit overkill.

I'm suggesting to revisit it when we are refactoring the monitor (renaming to usm, and easily supporting http2/kafka/ other protocols for decoding)

**Review by p-lambert — APPROVED** [review]

**Pull request merged**